src/main/java/fi/helsinki/cs/tmc/core/communication/TmcServerCommunicationTaskFactory.java

```diff
@@ -78,7 +81,33 @@ public void setSettings(TmcSettings settings) {
         this.settings = settings;
     }
 
-    private URI getCourseListUrl() {
+    /**
+     * Returns a Callable that calls the given Callable.
+     *
+     * <p>If the call fails, the auth is refreshed and the call is done again.</p>
+     *
+     * @param <T> return type of the callable
+     * @param callable Callable to be wrapped
+     * @return Callable that wraps the given callable
+     */
+    private <T> Callable<T> tryCallCallable(final Callable<T> callable) {
```

Comment:
Maybe give this method a name that makes it clear that the intention of this method is to call a callable with an oauth token.